### ISSUE TITLE
Error mounting '/dev/mapper/docker-202:1-*-init' on '/var/lib/docker/devicemapper/mnt/*-init': no such file or directory

### ISSUE THREAD

**Issue by kytwb** [docker/docker#9562]
Happens every now and then in Bamboo, using latest Docker:

```
08-Dec-2014 19:05:47	Step 5 : ENV NODE_ENV "production"
08-Dec-2014 19:05:47	 ---> Running in 84dc09338c80
08-Dec-2014 19:06:02	 ---> ad4d4bb7a6de
08-Dec-2014 19:06:04	Removing intermediate container 84dc09338c80
08-Dec-2014 19:06:04	Step 6 : EXPOSE 8001
08-Dec-2014 19:06:04	 ---> Running in 72ea6bc04404
08-Dec-2014 19:06:18	 ---> d0ff8da5ae27
08-Dec-2014 19:06:21	Removing intermediate container 72ea6bc04404
08-Dec-2014 19:06:21	Step 7 : CMD npm start
08-Dec-2014 19:06:21	 ---> Running in 6476a9fc45dc
08-Dec-2014 19:06:21	2014/12/08 12:06:21 Error mounting '/dev/mapper/docker-202:1-286333-6476a9fc45dcd2e2cbca7460678a868ffec2d94dd634f1a133f3dd5714f7edaf-init' on '/var/lib/docker/devicemapper/mnt/6476a9fc45dcd2e2cbca7460678a868ffec2d94dd634f1a133f3dd5714f7edaf-init': no such file or directory
08-Dec-2014 19:06:21	Failing task since return code of [/bin/sh /tmp/REAC-RCS-DB-186-ScriptBuildTask-7336754602013549492.sh] was 1 while expected 0
08-Dec-2014 19:06:21	Finished task 'Build Docker Image' with result: Failed
```

**Comment by vbatts** [comment]
@kytwb also, can you provide `docker info` and `docker version` from the build host that this fails on?

**Comment by ahromis** [comment]
@vbatts Wer are experiencing the same issue, always during the build process.  It happens rather sporadically though.  Our CI environment does a lot of building of containers and we see if maybe a few times a week.

error:

```
08-Jan-2015 11:53:46	Step 5 : COPY ./config/create_db.sh /docker-entrypoint-initdb.d/
08-Jan-2015 11:53:48	 ---> 7d0ee4cdc194
08-Jan-2015 11:53:49	Removing intermediate container 29402cc76c76
08-Jan-2015 11:53:49	Step 6 : EXPOSE 5432
08-Jan-2015 11:53:50	 ---> Running in 0543dcab43d8
08-Jan-2015 11:53:50	time="2015-01-08T11:53:50-05:00" level="info" msg="Error getting container 0543dcab43d873a968862b9a6dc958f450d6d4b82e715c503da85ff2740afb99 from driver devicemapper: Error mounting '/dev/mapper/docker-8:3-176658-0543dcab43d873a968862b9a6dc958f450d6d4b82e715c503da85ff2740afb99' on '/var/lib/docker/devicemapper/mnt/0543dcab43d873a968862b9a6dc958f450d6d4b82e715c503da85ff2740afb99': no such file or directory"
08-Jan-2015 11:53:50	Failing task since return code of [/bin/sh /tmp/TMS-TMS234-JOB1-4-ScriptBuildTask-448517969242182542.sh] was 1 while expected 0
08-Jan-2015 11:53:50	Finished task 'Build Database Container' with result: Failed
```

docker info:

```
Containers: 1
Images: 308
Storage Driver: devicemapper
 Pool Name: docker-8:3-268678259-pool
 Pool Blocksize: 65.54 kB
 Data file: /var/lib/docker/devicemapper/devicemapper/data
 Metadata file: /var/lib/docker/devicemapper/devicemapper/metadata
 Data Space Used: 8.804 GB
 Data Space Total: 64.42 GB
 Metadata Space Used: 16.02 MB
 Metadata Space Total: 17.05 GB
 Library Version: 1.02.82-git (2013-10-04)
Execution Driver: native-0.2
Kernel Version: 3.10.0-123.13.2.el7.x86_64
Operating System: CentOS Linux 7 (Core)
CPUs: 2
Total Memory: 3.704 GiB
Name: exp1cib11.nextgearcapital.com
ID: 3IZO:JEB2:6OTK:BD43:KW5P:MOOY:PG4T:7ZLP:V6DX:E53S:O4K2:HKX6
Username: nextgear
Registry: [https://index.docker.io/v1/]
```

docker version:

```
Client version: 1.4.1
Client API version: 1.16
Go version (client): go1.3.3
Git commit (client): 5bc2ff8
OS/Arch (client): linux/amd64
Server version: 1.4.1
Server API version: 1.16
Go version (server): go1.3.3
Git commit (server): 5bc2ff8
```

**Comment by IPvFletch** [comment]
I too have this problem using fabric and docker to build docker images. We're using Bamboo build server to run some bash scripts which run the fab cmds. It usually works, but occasionally we will get the error described above (/dev/mapper/... : no such file or dir).

As per @vbatts comments, I compiled docker-1.4.1-dev from the latest master branch. I am now using it for my Bamboo scripts, and will reply later on with any updates.

**Comment by IPvFletch** [comment]
So far, it looks like I occasionally get these errors still..
`Error response from daemon: Cannot start container b5460a52cb9138a03b729ba7a649c7686d140bbb9ca96c0cceb0fa29c74d131a: Error getting container b5460a52cb9138a03b729ba7a649c7686d140bbb9ca96c0cceb0fa29c74d131a from driver devicemapper: Error mounting '/dev/mapper/docker-202:80-4199007-b5460a52cb9138a03b729ba7a649c7686d140bbb9ca96c0cceb0fa29c74d131a' on '/vol/docker/devicemapper/mnt/b5460a52cb9138a03b729ba7a649c7686d140bbb9ca96c0cceb0fa29c74d131a': no such file or directory`

**Comment by IPvFletch** [comment]
`ls -al /var/lib/docker
lrwxrwxrwx 1 root root 11 Dec 15 17:28 /var/lib/docker -> /vol/docker`

From /etc/default/docker:
`DOCKER_OPTS="-g $(readlink -f /var/lib/docker)"`

**Comment by IPvFletch** [comment]
I switched from the default 'devicemapper' driver to 'aufs' and so far, no more problems..
`sudo apt-get install linux-image-extra-$(uname -r) aufs-tools`
`sudo service docker restart`

**Comment by jasonalmansor** [comment]
@IPvFletch unfortunately we can't since we're using CentOS... at least, not without a fair amount of work.

**Comment by vbatts** [comment]
Largely this issue sounds like a duplicate of #4036
For many hosts #10195 will help address this issue. If you rebuild on docker master, or wait on the upcoming docker-1.5 release, you can use this fix. Ensure that `docker info` includes the output `Udev Sync Supported: true`

**Comment by rjaquino** [comment]
We've downloaded the docker-1.5-rc4 binary, which we believe should include the commits in that pull request, yet we still see `Udev Sync Supported: false`. Is there anything else we need to do to ensure that that option is set correctly?

**Comment by jbruni** [comment]
I just got this "error mounting from driver devicemapper", for a simple `docker run -ti ubuntu:14.04 /bin/bash`, at a fresh AWS EC2 instance w/ Ubuntu 14.04 LTS image.

```
FATA[0000] Error response from daemon: Cannot start container 98c12b40f6d5fee7470e77bcb2e743ff15dd1707ff4a1eedd72b11e1142ba234: Error getting container 98c12b40f6d5fee7470e77bcb2e743ff15dd1707ff4a1eedd72b11e1142ba234 from driver devicemapper: Error mounting '/dev/mapper/docker-202:1-149822-98c12b40f6d5fee7470e77bcb2e743ff15dd1707ff4a1eedd72b11e1142ba234' on '/var/lib/docker/devicemapper/mnt/98c12b40f6d5fee7470e77bcb2e743ff15dd1707ff4a1eedd72b11e1142ba234': no such file or directory 
```

Docker version (omitted server because it is identical):
```
Client version: 1.4.1
Client API version: 1.16
Go version (client): go1.3.3
Git commit (client): 5bc2ff8
OS/Arch (client): linux/amd64
```

I tried the same `docker run` command immediately afterwards, and it just worked.

**Comment by IPvFletch** [comment]
J, Try enabling AUFS:

 

apt-get -y install linux-image-extra-$(uname -r) aufs-tools

apt-get -y install lxc-docker

**Comment by tiborvass** [comment]
Ping @vbatts @unclejack

**Comment by vbatts** [comment]
@rjaquino that is due to the static linked binary can _not_ sync udev. Udev is not statically linkable, therefore the docker static build has to exclude the flags to sync with udev. This is expected. There has been discussion about not compiling devicemapper in when building docker statically

**Comment by omribahumi** [comment]
@vbatts "make all" produces a statically linked binary. How do I make a dynamically linked one that supports Udev sync?

**Comment by vbatts** [comment]
@omribahumi I build the dynamic linked binary with `AUTO_GOPATH=1 ./project/make.sh dynbinary`. Key difference is that it builds _not_ in a container, but on the host, so all the dependent libraries will need to be installed

**Comment by omribahumi** [comment]
@vbatts I'm having trouble compiling this on Ubuntu 12.04 (Ubuntu 14.04 works fine).

The Ubuntu packages on "https://get.docker.com/ubuntu" are statically linked. I guess that's not going to change for 1.5, meaning #4036 will still happen and probably devicemapper support will be left out (#10664).

The only reason we switched from devicemapper to AUFS is #4036, I was hoping 1.5 will solve this for us.
We have other issues with AUFS.

**Comment by vbatts** [comment]
@omribahumi i'm not sure i have recommendations on ubuntu 12.04 ... since it may have a dated libdm (devicemapper) and udev from ubuntu. I know the sync functionality has gotten much better in newer releases (or heavily rebased releases like RHEL).

**Comment by omribahumi** [comment]
What about the packages at the official docker apt repository? Will it stay static?

**Comment by coderlol** [comment]
Is this error supposed to happen on 1.6.1 and Ubuntu 14.04.2 ?

INFO[0035] Error getting container 18414422ccb77bb21b8393fd3fb115225d889db1237c4496288a8e3b6b6f94d3 from driver devicemapper: Error mounting '/dev/mapper/docker-202:1-151847-18414422ccb77bb21b8393fd3fb115225d889db1237c4496288a8e3b6b6f94d3' on '/var/lib/docker/devicemapper/mnt/18414422ccb77bb21b8393fd3fb115225d889db1237c4496288a8e3b6b6f94d3': no such file or directory



ubuntu@ip-10-0-0-20:/data01/workarea/builder$ docker info
Containers: 5
Images: 82
Storage Driver: devicemapper
 Pool Name: docker-202:1-151847-pool
 Pool Blocksize: 65.54 kB
 Backing Filesystem: extfs
 Data file: /dev/loop0
 Metadata file: /dev/loop1
 Data Space Used: 2.528 GB
 Data Space Total: 107.4 GB
 Data Space Available: 3.994 GB
 Metadata Space Used: 4.317 MB
 Metadata Space Total: 2.147 GB
 Metadata Space Available: 2.143 GB
 Udev Sync Supported: false
 Data loop file: /var/lib/docker/devicemapper/devicemapper/data
 Metadata loop file: /var/lib/docker/devicemapper/devicemapper/metadata
 Library Version: 1.02.82-git (2013-10-04)
Execution Driver: native-0.2
Kernel Version: 3.13.0-48-generic
Operating System: Ubuntu 14.04.2 LTS
CPUs: 2
Total Memory: 3.859 GiB
Name: ip-10-0-0-20
ID: ZGV6:A7H6:QVL4:WDLD:LOWK:FOHK:6DV7:2LLL:BHBX:DGBX:HKF6:7HCH
WARNING: No swap limit support
ubuntu@ip-10-0-0-20:/data01/workarea/builder$ docker --version
Docker version 1.6.1, build 97cd073
ubuntu@ip-10-0-0-20:/data01/workarea/builder$

**Comment by mbentley** [comment]
@coderlol - You should either use AUFS support or compile docker as a dynamic binary as discussed above.

**Comment by igorescobar** [comment]
Same problem here...

```shell
Service 'build' failed to build: Error getting container 848c514c83ed558eaf8c2087c48888faaeb96c2e9a3d043bc9f25f815822550d from driver devicemapper: Error mounting '/dev/mapper/docker-202:1-524953-848c514c83ed558eaf8c2087c48888faaeb96c2e9a3d043bc9f25f815822550d' on '/var/lib/docker/devicemapper/mnt/848c514c83ed558eaf8c2087c48888faaeb96c2e9a3d043bc9f25f815822550d': no such file or directory
```
docker info
```shell
Containers: 5
Images: 522
Storage Driver: devicemapper
 Pool Name: docker-202:1-524953-pool
 Pool Blocksize: 65.54 kB
 Backing Filesystem: extfs
 Data file: /dev/loop0
 Metadata file: /dev/loop1
 Data Space Used: 13.14 GB
 Data Space Total: 107.4 GB
 Data Space Available: 94.24 GB
 Metadata Space Used: 23.07 MB
 Metadata Space Total: 2.147 GB
 Metadata Space Available: 2.124 GB
 Udev Sync Supported: false
 Deferred Removal Enabled: false
 Data loop file: /var/lib/docker/devicemapper/devicemapper/data
 Metadata loop file: /var/lib/docker/devicemapper/devicemapper/metadata
 Library Version: 1.02.82-git (2013-10-04)
Execution Driver: native-0.2
Logging Driver: json-file
Kernel Version: 3.13.0-48-generic
Operating System: Ubuntu 14.04.2 LTS
CPUs: 2
Total Memory: 3.676 GiB
Name: ip-172-31-27-40
ID: EKZJ:N3EL:CBI4:2IV3:MZLU:LYRQ:D7XG:3GAB:BJSR:6JTG:L32P:SQOG
WARNING: No swap limit support
```

docker version
```shell
Client version: 1.7.1
Client API version: 1.19
Go version (client): go1.4.2
Git commit (client): 786b29d
OS/Arch (client): linux/amd64
Server version: 1.7.1
Server API version: 1.19
Go version (server): go1.4.2
Git commit (server): 786b29d
OS/Arch (server): linux/amd64
```

**Comment by thaJeztah** [comment]
@igorescobar you're running devicemapper *without* udev sync support; as mentioned earlier in this thread this leads to many issues, and you really should not run Docker without udev sync. I suggest to;

- Install aufs and use the aufs driver, or
- Install a dynamically linked binary https://blog.docker.com/2015/07/new-apt-and-yum-repos/ and make sure you have a fully updated system to have recent libraries installed https://github.com/docker/docker/issues/9562#issuecomment-73702962

**Comment by igorescobar** [comment]
Thanks @thaJeztah !
We just upgraded our docker via ansible and everything seems to be working fine now. https://github.com/angstwad/docker.ubuntu

**Comment by toddlers** [comment]
@thaJeztah  thanks for the link :)
